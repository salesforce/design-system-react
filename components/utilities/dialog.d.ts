declare module '@salesforce/design-system-react/components/utilities/dialog' {
	import React from 'react';
	type Props = {
		/**
		 * Alignment of the dialog with respect to the target (assuming left-to-right language direction). For example,
		 * a value of 'left bottom' indicates that the dialog will be rendered below and left-aligned with the target.
		 * Note that setting the direction prop to "rtl" will flip the resulting dialog alignment.
		 */
		align?:
			| 'top'
			| 'top left'
			| 'top right'
			| 'right'
			| 'right top'
			| 'right bottom'
			| 'bottom'
			| 'bottom left'
			| 'bottom right'
			| 'left'
			| 'left top'
			| 'left bottom';
		/**
		 * CSS classes to be added to the absolutely positioned element.
		 */
		className?: any[] | Record<string, any> | string;
		/**
		 * CSS classes to be added to the wrapping `div` of the contents of the dialog.
		 */
		contentsClassName?: any[] | Record<string, any> | string;
		/**
		 * Contents of dialog
		 */
		children: React.ReactNode /*.isRequired*/;
		/**
		 * Closes dialog when tab key is pressed
		 */
		closeOnTabKey?: boolean;
		/**
		 * Props passed along to wrapping div. This allows one less wrapping `div` to be in the markup. dialog children are expected to be wrapper in a single `div`.
		 */
		containerProps?: Record<string, any>;
		/**
		 * Establishes directional context for component. Defaults to left-to-right.
		 */
		direction?: DIRECTIONS.LTR | DIRECTIONS.RTL;
		/**
		 * Will show the nubbin pointing from the dialog to the reference element. Positioning and offsets will be handled.
		 */
		hasNubbin?: boolean;
		/**
		 * By default, dialogs will flip their alignment (such as bottom to top) if they extend beyond a boundary element such as a scrolling parent or a window/viewpoint. `hasStaticAlignment` disables this behavior and allows this component to extend beyond boundary elements.
		 */
		hasStaticAlignment?: boolean;
		/**
		 * Sets the dialog width to the width of either 'target' (Menus attached to `input` typically follow this UX pattern), 'menu' or 'none.
		 */
		inheritWidthOf?: 'target' | 'menu' | 'none';
		/**
		 * DEPRECATED - do not add checkProp deprecation message at this level. It is handled at higher level components.
		 * TODO - to be removed.
		 * Offset adds pixels to the absolutely positioned dropdown menu in the format: ([vertical]px [horizontal]px). SHOULD BE OBJECT -----------
		 */
		offset?: string;
		/**
		 * Called when dialog closes and unmounts.
		 */
		onClose?: (v: any) => any;
		/**
		 * Called when a key pressed.
		 */
		onKeyDown?: (v: any) => any;
		/**
		 * Called when mouse hovers over the trigger button. This is only called if `this.props.openOn` is set to `hover`.
		 */
		onMouseEnter?: (v: any) => any;
		/**
		 * Called when mouse hover leaves the trigger button. This is only called if `this.props.openOn` is set to `hover`.
		 */
		onMouseLeave?: (v: any) => any;
		/**
		 * Called when dialog opens (that is mounts). The parameters are `undefined, { portal: this.portal }`.
		 */
		onOpen?: (v: any) => any;
		/**
		 * React component to be aligned with. Function should return a DOM `ref` from the parent component.
		 */
		onRequestTargetElement: (v: any) => any /*.isRequired*/;
		/**
		 * Triggered when an item in the menu is clicked.
		 */
		outsideClickIgnoreClass?: string;
		/**
		 * If a dialog is `positione="overflowBoundaryElement"`, it will be rendered in a portal or separate render tree. This `portalMount` callback will be triggered instead of the the default `ReactDOM.unstable_renderSubtreeIntoContainer` and the function will mount the portal itself. Consider the following code that bypasses the internal mount and uses an Enzyme wrapper to mount the React root tree to the DOM.
		 *
		 * ```
		 * <Popover
		 *   isOpen
		 *   portalMount={({ instance, reactElement, domContainerNode }) => {
		 *     portalWrapper = Enzyme.mount(reactElement, { attachTo: domContainerNode });
		 *   }}
		 *   onOpen={() => {
		 *     expect(portalWrapper.find(`#my-heading`)).to.exist;
		 *     done();
		 *   }}
		 *   />
		 * ```
		 */
		portalMount?: (v: any) => any;
		/**
		 * Please select one of the following:
		 * * `absolute` - (default) The dialog will use `position: absolute` and style attributes to position itself. This allows inverted placement or flipping of the dialog.
		 * * `overflowBoundaryElement` - The dialog will overflow scrolling parents. Use on elements that are aligned to the left or right of their target and don't care about the target being within a scrolling parent. Typically this is a popover or tooltip. Dropdown menus can usually open up and down if no room exists. In order to achieve this a portal element will be created and attached to `body`. This element will render into that detached render tree.
		 * * `relative` - No styling or portals will be used. Menus will be positioned relative to their triggers. This is a great choice for HTML snapshot testing.
		 */
		position?:
			| 'absolute'
			| 'overflowBoundaryElement'
			| 'relative' /*.isRequired*/;
		/**
		 * An object of CSS styles that are applied to the immediate parent `div` of the contents. Use this instead of margin props.
		 */
		style?: Record<string, any>;

		/**
		 * Sets which focus UX pattern to follow. For instance, popovers trap focus and must be exited to regain focus. Dropdowns and Tooltips never have focus.
		 */
		variant?: 'dropdown' | 'popover' | 'tooltip';
	};
	/*
	 * A Dialog is content that is separate from the typical flow of a page. It typically overlays other elements in the document flow. This is achieved with elevation (`z-index`) and one of the following: relative position, absolute position, or a new top-level React render tree (portal). A boundary element is a scrolling ancestor element or the edge of the browser (window/viewport). This element typically has an overflow (overflow-y/overflow-x) style that is scroll, hidden, or auto. Inverted placement is the flipping of the overlay element from top to bottom or left to right in order stay within a boundary element.
	 *
	 * * Dropdown menu (Combobox, DatePicker, et al.) placement is typically bottom-aligned and should be allowed to invert its placement when inside a boundary element this often happens within a modal. Dropdowns should not overflow boundary elements , since most boundary elements scroll vertically and have space for the menu.
	 *
	 * * If they are hidden, left and right placed overlay elements (such as Popover and Tooltip) should be placed within a portal element attached to the DOM <body>, but styled to align to its target/trigger. Since scrolling typically occurs on the vertical axis, this allows them to overflow boundary elements and still allow scrolling of content, yet still invert placement for the browser viewport. Portal elements are only necessary if an original ancestor boundary element exists. **No portals are created by default.**
	 *
	 * * Nubbins/arrows should be repositioned for any new placement/alignment.
	 *
	 * Allow Overflowing of Boundary Element: Allow applications to create a portal element attached to the DOM <body> to be outside of boundary elements if manual testing shows confusing alignment/poor usability/readability. This should be exception and not the default.
	 *
	 * Allow Inverted placement: Allow applications to prevent inverted placement if manual testing shows confusing alignment/poor usability/readability.
	 *
	 * ### How this new Dialog component works
	 * * There is no longer an inline render within components. All overlays should go through `Dialog`. `position: relative` just passes on the markup with some additional event listeners. No positional library is used.
	 * * The default `position: absolute` will run through three renders. It will first render the overlay at `0px/0px` offset of its parent. It will then store the target DOM node, once the DOM node is set, a state change will occur and tell a PopperJS instance to be created on update, once it is created, a third render is done to update the styles created by PopperJS.
	 * * `position: overflowBoundaryElement` will do the same three renders as `absolute` except that the initial render will create a disconnected render tree (portal) on the `body`. Then, the position will change once the target is stored. The portal itself will be rendered multiple times. The first will result in `onOpen` executing. Each update will result in a re-render of the disconnected render-tree.
	 *
	 * This component is private.
	 */
	function Component(props: Props): JSX.Element;
	export default Component;
}
