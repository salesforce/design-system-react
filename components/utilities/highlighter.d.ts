declare module '@salesforce/design-system-react/components/utilities/highlighter' {
	import React from 'react';
	type Props = {
		/**
		 * The full string to display.
		 */
		children?: string | number | boolean | React.ReactNode;
		className?: string;
		/**
		 * The string of text (or Regular Expression) to highlight.
		 */
		search?: PropTypes.any;
	};

	function Component(props: Props): JSX.Element;
	export default Component;
}
