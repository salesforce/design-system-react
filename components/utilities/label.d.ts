declare module '@salesforce/design-system-react/components/utilities/label' {
	import React from 'react';
	type Props = {
		/*
		 * Assistive Text to use instead of a visible label
		 */
		assistiveText?: Record<string, any>;
		/**
		 * Class names to be added to the label
		 */
		className?: any[] | Record<string, any> | string;
		/*
		 * Id of the input associated with this label
		 */
		htmlFor?: string;
		/*
		 * Input Label
		 */
		label?: string;
		/*
		 * Applies label styling for a required form element
		 */
		required?: boolean;
		/**
		 * Changes markup of label.
		 */
		variant?: 'base' | 'static';
	};

	function Component(props: Props): JSX.Element;
	export default Component;
}
