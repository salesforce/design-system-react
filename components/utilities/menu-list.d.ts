declare module '@salesforce/design-system-react/components/utilities/menu-list' {
	import React from 'react';
	type Props = {
		/**
		 * Determines whether or not to show a checkmark for selected items.
		 */
		checkmark?: boolean;
		/**
		 * CSS classes to be added to `<ul />`.
		 */
		className?: string;
		/**
		 * Used internally to determine the id that will be used for list items.
		 */
		getListItemId?: (v: any) => any;
		/**
		 * Used internally to pass references to the individual menu items back up for focusing / scrolling.
		 */
		itemRefs?: (v: any) => any;
		/**
		 * If provided, this function will be used to render the contents of each menu item.
		 */
		itemRenderer?: (v: any) => any;
		/**
		 * Sets the height of the list based on the numeber of items.
		 */
		length?: null | '5' | '7' | '10' | 5 | 7 | 10;
		/**
		 * Triggered when a list item is selected (via mouse or keyboard).
		 */
		onSelect?: (v: any) => any;
		/**
		 * An array of items to render in the list.
		 */
		options?: any[];
		/**
		 * The index of the currently selected item in the list.
		 */
		selectedIndex?: number;
		/**
		 * Accepts a `Tooltip` component to be used as the template for menu item tooltips that appear via the `tooltipContent` options object attribute
		 */
		tooltipMenuItem?: React.ReactNode;
		/**
		 * The id of the element which triggered this list (in a menu context).
		 */
		triggerId?: string;
	};
	/**
	 * Component description.
	 */
	function Component(props: Props): JSX.Element;
	export default Component;
}
