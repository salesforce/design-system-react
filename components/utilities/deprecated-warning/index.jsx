/* Copyright (c) 2015-present, salesforce.com, inc. All rights reserved */
/* Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license */

// ### React
import React from 'react';

// ## Constants
import { DEPRECATED_WARNING } from '../../../utilities/constants';

import Alert from '../../alert';
import AlertContainer from '../../alert/container';

/**
 * A utility component that is used to highlight a deprecated component.
 */
const DeprecatedWarning = () => (
	<div className="slds-p-top_x-large">
		<AlertContainer>
			<Alert
				labels={{
					heading: (
						<span>
							<strong>Deprecated component.</strong> No fit and finish bug fixes
							will be accepted for this component.
						</span>
					),
				}}
				variant="error"
			/>
		</AlertContainer>
	</div>
);

// ### Display Name
DeprecatedWarning.displayName = DEPRECATED_WARNING;

export default DeprecatedWarning;
