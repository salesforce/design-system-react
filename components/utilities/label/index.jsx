/* eslint-disable react/jsx-curly-brace-presence */
/* Copyright (c) 2015-present, salesforce.com, inc. All rights reserved */
/* Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license */

import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

const propTypes = {
	/*
	 * Assistive Text to use instead of a visible label
	 */
	assistiveText: PropTypes.object,
	/**
	 * Class names to be added to the label
	 */
	className: PropTypes.oneOfType([
		PropTypes.array,
		PropTypes.object,
		PropTypes.string,
	]),
	/*
	 * Id of the input associated with this label
	 */
	htmlFor: PropTypes.string,
	/*
	 * Input Label
	 */
	label: PropTypes.string,
	/*
	 * Applies label styling for a required form element
	 */
	required: PropTypes.bool,
	/**
	 * Changes markup of label.
	 */
	variant: PropTypes.oneOf(['base', 'static']),
};

/*
 * Form label. This returns null if there is no label text (hidden or shown)
 */
const Label = ({
	variant = 'base',
	label,
	assistiveText,
	className,
	htmlFor,
	required,
}) => {
	const labelText = label || (assistiveText && assistiveText.label); // One of these is required to pass accessibility tests

	const subRenders = {
		base: (
			<label
				className={classNames(
					'slds-form-element__label',
					{
						'slds-assistive-text': assistiveText && !label,
					},
					className
				)}
				htmlFor={htmlFor}
			>
				{required && (
					<abbr className="slds-required" title="required">
						{'*'}
					</abbr>
				)}
				{labelText}
			</label>
		),
		static: (
			<span className={classNames('slds-form-element__label', className)}>
				{labelText}
			</span>
		),
	};

	return labelText ? subRenders[variant] : null;
};

Label.displayName = 'Label';
Label.propTypes = propTypes;

export default Label;
