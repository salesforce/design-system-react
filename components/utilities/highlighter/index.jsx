/* Copyright (c) 2015-present, salesforce.com, inc. All rights reserved */
/* Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license */

// ### React
import React from 'react';
import PropTypes from 'prop-types';

// ### ReactHighlighter
import ReactHighlighter from 'react-highlighter';

// ## Constants
import { HIGHLIGHTER } from '../../../utilities/constants';

/**
 * A utility component that highlights occurrences of a particular pattern in its contents.
 */
const Highlighter = (props) => {
	if (props.search) {
		let children;
		if (typeof props.children === 'string') {
			children = (
				<ReactHighlighter
					className={props.className}
					matchClass={null}
					matchElement="mark"
					search={props.search}
					title={props.children}
				>
					{props.children}
				</ReactHighlighter>
			);
		} else {
			const findString = (nodeArr) =>
				nodeArr.map((element) => {
					let newElement;
					if (typeof element === 'string') {
						newElement = (
							<ReactHighlighter
								key={element}
								className={props.className}
								matchClass={null}
								matchElement="mark"
								search={props.search}
								title={element}
							>
								{element}
							</ReactHighlighter>
						);
					} else {
						newElement = element;
					}
					return newElement;
				});

			if (props.children.props) {
				const node = props.children.props.children;
				children = node instanceof Array ? findString(node) : node;
			}
		}

		return <span>{children}</span>;
	}

	if (typeof props.children === 'string') {
		return (
			<span className={props.className} title={props.children}>
				{props.children}
			</span>
		);
	}

	return <span className={props.className}>{props.children}</span>;
};

// ### Display Name
Highlighter.displayName = HIGHLIGHTER;

// ### Prop Types
Highlighter.propTypes = {
	/**
	 * The full string to display.
	 */
	children: PropTypes.oneOfType([
		PropTypes.string,
		PropTypes.number,
		PropTypes.bool,
		PropTypes.node,
	]),
	className: PropTypes.string,
	/**
	 * The string of text (or Regular Expression) to highlight.
	 */
	search: PropTypes.any,
};

export default Highlighter;
