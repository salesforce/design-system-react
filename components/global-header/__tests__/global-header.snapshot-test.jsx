import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import {
	testDOMandHTML,
	testImageSnapshot
} from '../../../tests/snapshot-helpers';

import Default from '../__examples__/default';

import { GLOBAL_HEADER } from '../../../utilities/constants';

describe(GLOBAL_HEADER, () => {
	// test('Default DOM & HTML Snapshots look the same', () => {
	// 	testDOMandHTML(Default);
	// });

	test('Basic (Fluid Layout) Image Snapshot looks the same', async () => {
		await testImageSnapshot(GLOBAL_HEADER, 'Basic (Fluid Layout)');
	});

	test('Advanced (Fixed Layout) Image Snapshot looks the same', async () => {
		await testImageSnapshot(GLOBAL_HEADER, 'Advanced (Fixed Layout)');
	});
});
