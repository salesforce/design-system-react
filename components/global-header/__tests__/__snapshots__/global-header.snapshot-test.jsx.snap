// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SLDSGlobalHeader Default DOM & HTML Snapshots look the same 1`] = `
<header
  className="slds-global-header_container"
>
  <a
    className="slds-assistive-text slds-assistive-text--focus"
    href="javascript:void(0);"
    onClick={[Function]}
  >
    Skip to Navigation
  </a>
  <a
    className="slds-assistive-text slds-assistive-text--focus"
    href="javascript:void(0);"
    onClick={[Function]}
  >
    Skip to Main Content
  </a>
  <div
    className="slds-global-header slds-grid slds-grid--align-spread"
  >
    <div
      className="slds-global-header__item"
    >
      <div
        className="slds-global-header__logo"
      >
        <img
          alt=""
          src="/images/logo.svg"
        />
      </div>
    </div>
    <div
      className="slds-global-header__item slds-global-header__item--search"
    >
      <div
        className="slds-form-element slds-lookup"
        data-scope="single"
        data-select="single"
        onScroll={undefined}
      >
        <div
          className="slds-form-element__control slds-input-has-icon slds-input-has-icon--left"
        >
          <div
            className="slds-form-element slds-lookup__search-input"
          >
            <div
              className="slds-form-element__control slds-input-has-icon slds-input-has-icon_right"
            >
              <input
                aria-activedescendant=""
                aria-autocomplete="list"
                aria-controls={undefined}
                aria-describedby={undefined}
                aria-expanded={false}
                aria-labelledby={undefined}
                aria-owns={undefined}
                aria-required={undefined}
                autoComplete={undefined}
                className="slds-input"
                disabled={undefined}
                id="undefinedLookup"
                maxLength={undefined}
                minLength={undefined}
                name={undefined}
                onBlur={[Function]}
                onChange={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onInput={undefined}
                onInvalid={undefined}
                onKeyDown={[Function]}
                onKeyPress={undefined}
                onKeyUp={undefined}
                onSelect={undefined}
                onSubmit={undefined}
                placeholder="Search Salesforce"
                readOnly={undefined}
                required={undefined}
                role="combobox"
                style={undefined}
                tabIndex={undefined}
                type="text"
                value=""
              />
              <svg
                aria-hidden={true}
                className="slds-input__icon slds-icon-text-default slds-input__icon--right"
              >
                <use
                  xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#search"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
    </div>
    <ul
      className="slds-global-header__item slds-grid slds-grid--vertical-align-center"
    >
      <li>
        <button
          aria-controls={undefined}
          aria-describedby={undefined}
          aria-expanded={undefined}
          aria-haspopup={undefined}
          className="slds-button slds-button--neutral slds-m-right--small"
          disabled={false}
          id={undefined}
          onBlur={undefined}
          onClick={[Function]}
          onFocus={undefined}
          onKeyDown={undefined}
          onKeyPress={undefined}
          onKeyUp={undefined}
          onMouseDown={undefined}
          onMouseEnter={undefined}
          onMouseLeave={undefined}
          tabIndex={undefined}
          title={undefined}
          type="button"
        >
          Feedback
        </button>
      </li>
      <li
        aria-haspopup="true"
        className="slds-dropdown-trigger slds-dropdown-trigger--click ignore-click-HJedWQ7a4G"
        id="HJedWQ7a4G"
        onBlur={undefined}
        onClick={[Function]}
        onFocus={null}
        onKeyDown={[Function]}
        onMouseDown={undefined}
        onMouseEnter={null}
        onMouseLeave={null}
      >
        <li>
          <button
            aria-controls={undefined}
            aria-describedby={undefined}
            aria-expanded={undefined}
            aria-haspopup={true}
            className="slds-button slds-button--icon-container slds-global-header__button--icon"
            disabled={false}
            id={undefined}
            onBlur={undefined}
            onClick={[Function]}
            onFocus={undefined}
            onKeyDown={undefined}
            onKeyPress={undefined}
            onKeyUp={undefined}
            onMouseDown={undefined}
            onMouseEnter={undefined}
            onMouseLeave={undefined}
            tabIndex="0"
            title={undefined}
            type="button"
          >
            <svg
              aria-hidden="true"
              className="slds-button__icon slds-global-header__icon"
            >
              <use
                xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#add"
              />
            </svg>
            <span
              className="slds-assistive-text"
            >
              Global Actions
            </span>
          </button>
        </li>
      </li>
      <li>
        <button
          aria-controls={undefined}
          aria-describedby={undefined}
          aria-expanded={undefined}
          aria-haspopup={undefined}
          className="slds-button slds-button--icon-container slds-global-header__button--icon"
          disabled={false}
          id={undefined}
          onBlur={undefined}
          onClick={[Function]}
          onFocus={undefined}
          onKeyDown={undefined}
          onKeyPress={undefined}
          onKeyUp={undefined}
          onMouseDown={undefined}
          onMouseEnter={undefined}
          onMouseLeave={undefined}
          tabIndex={undefined}
          title={undefined}
          type="button"
        >
          <svg
            aria-hidden="true"
            className="slds-button__icon slds-global-header__icon"
          >
            <use
              xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#question"
            />
          </svg>
          <span
            className="slds-assistive-text"
          >
            Help and Training
          </span>
        </button>
      </li>
      <li>
        <button
          aria-controls={undefined}
          aria-describedby={undefined}
          aria-expanded={undefined}
          aria-haspopup={undefined}
          className="slds-button slds-button--icon-container slds-global-header__button--icon"
          disabled={false}
          id={undefined}
          onBlur={undefined}
          onClick={[Function]}
          onFocus={undefined}
          onKeyDown={undefined}
          onKeyPress={undefined}
          onKeyUp={undefined}
          onMouseDown={undefined}
          onMouseEnter={undefined}
          onMouseLeave={undefined}
          tabIndex={undefined}
          title={undefined}
          type="button"
        >
          <svg
            aria-hidden="true"
            className="slds-button__icon slds-global-header__icon"
          >
            <use
              xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#settings"
            />
          </svg>
          <span
            className="slds-assistive-text"
          >
            Setup
          </span>
        </button>
      </li>
      <li
        aria-haspopup="true"
        className="slds-dropdown-trigger slds-dropdown-trigger--click slds-m-left--x-small"
        id="HJ-d-Q7pVz"
        onBlur={undefined}
        onClick={[Function]}
        onFocus={null}
        onKeyDown={[Function]}
        onMouseDown={undefined}
        onMouseEnter={null}
        onMouseLeave={null}
      >
        <li>
          <button
            aria-controls={undefined}
            aria-describedby={undefined}
            aria-expanded={undefined}
            aria-haspopup={true}
            className="slds-button slds-button--icon-container"
            disabled={false}
            id={undefined}
            onBlur={undefined}
            onClick={[Function]}
            onFocus={undefined}
            onKeyDown={undefined}
            onKeyPress={undefined}
            onKeyUp={undefined}
            onMouseDown={undefined}
            onMouseEnter={undefined}
            onMouseLeave={undefined}
            tabIndex="0"
            title={undefined}
            type="button"
          >
            <span
              className="slds-avatar slds-avatar--circle slds-avatar--medium"
            >
              <img
                alt=""
                src="/images/avatar2.jpg"
              />
            </span>
          </button>
        </li>
      </li>
    </ul>
  </div>
</header>
`;

exports[`SLDSGlobalHeader Default DOM & HTML Snapshots look the same 2`] = `
"<header class=\\"slds-global-header_container\\"><a href=\\"javascript:void(0);\\" class=\\"slds-assistive-text slds-assistive-text--focus\\">Skip to Navigation</a><a href=\\"javascript:void(0);\\" class=\\"slds-assistive-text slds-assistive-text--focus\\">Skip to Main Content</a>
  <div class=\\"slds-global-header slds-grid slds-grid--align-spread\\">
    <div class=\\"slds-global-header__item\\">
      <div class=\\"slds-global-header__logo\\"><img src=\\"/images/logo.svg\\" alt=\\"\\" /></div>
    </div>
    <div class=\\"slds-global-header__item slds-global-header__item--search\\">
      <div class=\\"slds-form-element slds-lookup\\" data-select=\\"single\\" data-scope=\\"single\\">
        <div class=\\"slds-form-element__control slds-input-has-icon slds-input-has-icon--left\\">
          <div class=\\"slds-form-element slds-lookup__search-input\\">
            <div class=\\"slds-form-element__control slds-input-has-icon slds-input-has-icon_right\\"><input type=\\"text\\" aria-activedescendant=\\"\\" aria-autocomplete=\\"list\\" aria-expanded=\\"false\\" class=\\"slds-input\\" id=\\"undefinedLookup\\" placeholder=\\"Search Salesforce\\" role=\\"combobox\\" value=\\"\\" /><svg aria-hidden=\\"true\\" class=\\"slds-input__icon slds-icon-text-default slds-input__icon--right\\"><use xlink:href=\\"/assets/icons/utility-sprite/svg/symbols.svg#search\\"></use></svg></div>
          </div>
        </div>
      </div>
    </div>
    <ul class=\\"slds-global-header__item slds-grid slds-grid--vertical-align-center\\">
      <li><button class=\\"slds-button slds-button--neutral slds-m-right--small\\" type=\\"button\\">Feedback</button></li>
      <li aria-haspopup=\\"true\\" class=\\"slds-dropdown-trigger slds-dropdown-trigger--click ignore-click-Sy7uZQ7a4M\\" id=\\"Sy7uZQ7a4M\\">
        <li><button aria-haspopup=\\"true\\" class=\\"slds-button slds-button--icon-container slds-global-header__button--icon\\" tabindex=\\"0\\" type=\\"button\\"><svg aria-hidden=\\"true\\" class=\\"slds-button__icon slds-global-header__icon\\"><use xlink:href=\\"/assets/icons/utility-sprite/svg/symbols.svg#add\\"></use></svg><span class=\\"slds-assistive-text\\">Global Actions</span></button></li>
      </li>
      <li><button class=\\"slds-button slds-button--icon-container slds-global-header__button--icon\\" type=\\"button\\"><svg aria-hidden=\\"true\\" class=\\"slds-button__icon slds-global-header__icon\\"><use xlink:href=\\"/assets/icons/utility-sprite/svg/symbols.svg#question\\"></use></svg><span class=\\"slds-assistive-text\\">Help and Training</span></button></li>
      <li><button class=\\"slds-button slds-button--icon-container slds-global-header__button--icon\\" type=\\"button\\"><svg aria-hidden=\\"true\\" class=\\"slds-button__icon slds-global-header__icon\\"><use xlink:href=\\"/assets/icons/utility-sprite/svg/symbols.svg#settings\\"></use></svg><span class=\\"slds-assistive-text\\">Setup</span></button></li>
      <li
        aria-haspopup=\\"true\\" class=\\"slds-dropdown-trigger slds-dropdown-trigger--click slds-m-left--x-small\\" id=\\"Hy4uZXmTNf\\">
        <li><button aria-haspopup=\\"true\\" class=\\"slds-button slds-button--icon-container\\" tabindex=\\"0\\" type=\\"button\\"><span class=\\"slds-avatar slds-avatar--circle slds-avatar--medium\\"><img src=\\"/images/avatar2.jpg\\" alt=\\"\\"/></span></button></li>
        </li>
    </ul>
  </div>
</header>"
`;
